File: linear_search/python/linear_search.py
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

if __name__ == "__main__":
    arr = [3, 45, 1, 2, 99]
    target = 1 

    result_index = linear_search(arr, target)
    if result_index != -1:
        print(f"Element found at index: {result_index}")
    else:
        print("Element not found in array")

File: 04_linear_search/java-code/LinearSearch.java
public class LinearSearch {
    public static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] arr = { 3, 45, 1, 2, 99 };
        int target = 1;
        int resultIndex = linearSearch(arr, target);
        if (resultIndex != -1) {
            System.out.println("Element found at index: " + resultIndex);
        } else {
            System.out.println("Element not found in the array");
        }
    }
}
